### ISSUE TITLE
Using `evil-paste-from-register` in ivy

### ISSUE THREAD

**Issue by ChoppinBlockParty** [emacs-evil/evil#1097]
# Question

# Environment
Evil search module: evil-search
Emacs version: 26.1
Operating System: Ubuntu 16.04
Evil version: 1.2.13
Evil installation type: MELPA
Graphical/Terminal: X
Tested in a `make emacs` session (see [CONTRIBUTING.md](https://github.com/emacs-evil/evil/blob/master/CONTRIBUTING.md#bug-reports)): Yes

I am trying to use `evil-paste-from-register` in ivy-mode. It seems work fine when I use register, e.g. `*`, `/`, `#`, etc. However, `C-w` (insert current word) is not working, nothing happens. Any ideas how I can make it work?

**Comment by mohkale** [comment]
I'm facing the same sort of issue, sometime I'd like to paste the current buffers file name into the ivy dialog (eg. when I'm moving it to another directory, or wish to use the same name but add a suffix).

I've found that `evil-get-register` uses `evil-ex-current-buffer` to determine where it gets such variables from. if we can find some way to set that variable to `(ivy-state-buffer ivy-last)` then it should work. I just can't find a nice way to hook into the ivy ecosystem to achieve that.

I tried:

```elisp
(advice-add 'ivy--minibuffer-setup :after
            (lambda ()
              (setq evil-ex-current-buffer (ivy-state-buffer ivy-last))))

(advice-add 'ivy--minibuffer-cleanup :after
               (lambda ()
                 (setq evil-ex-current-buffer nil)))
```

but that didn't work 😢.
